NewEmuChecklist.txt

```diff
@@ -0,0 +1,75 @@
+RetroAchievements Emulator Requirements:
+* If the emulator does not support a feature, the requirement is automatically met. This is primarily a set of
+  guidelines to ensure the interaction between the emulator and the toolkit is consistent between emulators.
+* If the system is supported by RetroArch, explain why a standalone emulator is desirable. RetroArch support is 
+  preferred as it is usable by more players (i.e. not just Windows users), and adding support to RALibRetro for 
+  an additional core is much easier than overhauling an entire emulator.
+
+[ ] Game Identification
+  [ ] Hash method documented and approved
+  [ ] If emulator supports dynamic patching [i.e. ips], hash is calculated after applying patch
+  [ ] Hash ignores any persistant save data [SRAM]
+  [ ] Game can be properly identified (RA_OnLoadNewRom)
+  [ ] Unidentified game provides some descriptor [typically filename] to unknown game dialog (RA_InstallSharedFunctions>RA_GetEstimatedGameTitle)
+  [ ] Unidentified game provides list of known games for console (RA_SetConsoleID)
+[ ] Emulator
+  [ ] DLL is downloaded if not available (RA_Init - new enum may be needed)
+  [ ] RetroAchievements menu is visible (RA_CreatePopupMenu) and functional (RA_InvokeDialog)
```

Comment:
I believe that this is all handled by `RAIntegration`.

Reply:
nope, RA_CreatePopupMenu returns a handle that must be added to the application's menu bar: https://github.com/RetroAchievements/RAEmus/blob/master/RAQUASI88/src/WIN32/retroachievements.cpp#L141

And RA_InvokeDialog must be called from the applications WndProc: https://github.com/RetroAchievements/RAEmus/blob/master/RAQUASI88/src/WIN32/retroachievements.cpp#L334

Reply:
> RA_CreatePopupMenu returns a handle that must be added to the application's menu bar

I see, must have missed it while copying and pasting this function. :)